Standards/scs-0114-v1-secure-connections.md

```diff
@@ -0,0 +1,172 @@
+  - If using RabbitMQ, the OpenStack services' oslo.messaging configuration for RabbitMQ MUST use the `kombu_ssl_*` options accordingly to enable SSL.
+  - If using Qpid, the OpenStack services' oslo.messaging configuration for Qpid MUST set the `qpid_protocol` option to `ssl` to enable SSL.
+- If using Apache Kafka, the server listener MUST be configured to accept SSL connections. See [Apache Kafka Listener Configuration](https://kafka.apache.org/documentation/#listener_configuration).
+  - The OpenStack services' oslo.messaging configuration for Kafka MUST specify `security_protocol` as either `SSL` or `SASL_SSL` and the related options appropriately. See [Kafka Driver Options](https://docs.openstack.org/oslo.messaging/latest/admin/kafka.html#driver-options).
+
+### Hypervisor and Live Migration Connections
+
+- If using libvirt on compute nodes, the libvirt port (as per `listen_addr`) MUST NOT be exposed to the network in an unauthenticated and unprotected fashion. It SHOULD be bound to `127.0.0.1`.
+- If QEMU and libvirt are used as the hypervisor interface in Nova, QEMU-native TLS SHOULD be used. See [Secure live migration with QEMU-native TLS](https://docs.openstack.org/nova/latest/admin/secure-live-migration-with-qemu-native-tls.html).
+
+
+### External VM Connections
+
+- As an OPTIONAL measure to assist customers in protecting external connections to their OpenStack networks and VMs, the Neutron VPNaaS service MAY be integrated into the infrastructure and the Neutron VPNaaS API extension offered to users. See the [Neutron VPNaaS documentation](https://docs.openstack.org/neutron-vpnaas/latest/).
+
+### Internal Neutron Connections
+
+- As an OPTIONAL measure to protect Neutron SDN traffic between physical nodes within the infrastructure, IPsec tunnels MAY be established between nodes involved in Neutron networks, such as compute and network controller nodes, at the network interfaces configured in Neutron.
+
+## Related Documents
+
+- [OpenStack Security Guide](https://docs.openstack.org/security-guide/)
+  - [OpenStack Security Guide: Secure communication](https://docs.openstack.org/security-guide/secure-communication.html)
+  - [OpenStack Security Guide: Database transport security](https://docs.openstack.org/security-guide/databases/database-transport-security.html)
+  - [OpenStack Security Guide: Messaging transport security](https://docs.openstack.org/security-guide/messaging/security.html#messaging-transport-security)
+- [Nova Documentation: Secure live migration with QEMU-native TLS](https://docs.openstack.org/nova/latest/admin/secure-live-migration-with-qemu-native-tls.html)
+- [Guide to TLS Standards Compliance](https://www.ssl.com/guide/tls-standards-compliance/)
+
+## Conformance Tests
+
+Conformance tests are limited to communication channels exposed to users, such as the public API interfaces.
+
```

Comment:
What would be necessary to test here besides the exposed channels?
RPC and DB? Maybe Libvirt? - Can we audit this somehow? Maybe an tcp-dump looking for such traffic?

However we will decide on this - it should also be documented here, so CSPs will know how to deal with possible audits.

Reply:
Essentially everything that mentions a "MUST" or "MUST NOT" in this standard, except for the external API.

- AMQP/RPC
- SQL/DB
- internal API
- Libvirt port

Most of it could be determined based on a network traffic recording (e.g. tcpdump) by looking at the port numbers and traffic in conjunction with the target addresses/ports configured in the OpenStack `*.conf`s. For the Libvirt port exposure, a `netstat` on the compute nodes might help or simply `grep`'ing the Libvirt configuration for `listen_addr`.

Most of this is deep in the infrastructure though. A filtered network capture might work but I don't think CSPs would be especially happy about it.

Is there any existing standard where non-external audit is proposed? I'm unsure how to tackle this in the standard.

Reply:
At least, standard should mention, that conformance tests do only check external endpoints. Internal endpoints MUST be checked with manual audits. We will have manual audits in other conformance tests as well. E.g. in #555, we investigate, how to check flavor implementation and I guess, manual audits are required in this context, too. In favor to of this PR, I suggest to open a new issue to propose a general concept for manual audits.

Reply:
I agree with Anja, we need a central place for manual audits. And a document that includes all necessary checks, so any auditor can simply follow through.

Reply:
> At least, standard should mention, that conformance tests do only check external endpoints. Internal endpoints MUST be checked with manual audits.

I added more emphasis on this with an additional sentence.

> In favor to of this PR, I suggest to open a new issue to propose a general concept for manual audits.

> I agree with Anja, we need a central place for manual audits.

https://github.com/SovereignCloudStack/standards/issues/611